lib/request.coffee

```diff
@@ -97,6 +103,32 @@ module.exports = getRequest = ({ token, debug = false, retries = 0, isBrowser =
 
 			return options
 
+	interceptRequestOptions = (requestOptions) ->
+		interceptRequestOrError(Promise.resolve(requestOptions))
+
+	interceptRequestError = (requestError) ->
+		interceptRequestOrError(Promise.reject(requestError))
+
+	interceptResponse = (response) ->
+		interceptResponseOrError(Promise.resolve(response))
+
+	interceptResponseError = (responseError) ->
+		interceptResponseOrError(Promise.reject(responseError))
+
+	interceptRequestOrError = (initialPromise) ->
+		Promise.resolve(
```

Comment:
I think this the result of reduce could be a non-bluebird promise, if a hook internally returns a native promise - this makes sure we definitely always get a bluebird promise at the end.